apps/site/pages/en/learn/asynchronous-work/understanding-processnexttick.md

```diff
@@ -51,4 +51,13 @@ Running before the timeout => number 3
 The timeout running last => number 4
 ```
 
-The exact output may differ from run to run.
+The output aforementioned holds true in ES Modules, e.g. mjs files, but keep in mind in CommonJS case, the output may be different:
+
+```bash
+Hello => number 1
+Running at next tick => number 2
+The timeout running last => number 4
+Running before the timeout => number 3
+```
+
+This is because the execution order of setImmediate and setTimeout is undeterministic in CommonJs.
```

Comment:
I don't think that's true, AFAIK CJS and ESM exposes the exact same `setImmediate` and `setTimeout` functions.
```suggestion
This is because the execution order of `setImmediate` and `setTimeout` is undeterministic in CommonJS.
```

Reply:
Suggestion implemented. They may expose same function but it's deterministic in modules because of asynchronous nature.